File: python/03/part1.py
def vals(x):
    if x.isupper():
        return ord(x)-38
    else:
        return ord(x)-96


def part1(x):
    with open(x, "r") as f:
        data = [[pack[0:len(pack)//2], pack[len(pack)//2:]]
                for pack in f.read().splitlines()]
        return sum((vals(list(set(i[0]) & set(i[1]))[0]))
                   for i in data)


test_val = 157
if part1("test") == test_val:
    print('Solution is {}'.format(part1("input")))
else:
    print("Test failed")

File: rust/03/src/part1.rs
use std::collections::HashSet;

fn char_to_val(c: char) -> u32 {
    match c {
        'a'..='z' => c as u32 - 'a' as u32 + 1,
        'A'..='Z' => c as u32 - 'A' as u32 + 27,
        _ => 0,
    }
}

fn solution(_data: String) -> u32 {
    let mut total = 0;
    let _val = _data.lines().map(|line| line).for_each(|line| {
        let (s1, s2) = line.split_at(line.len() / 2);
        let set: HashSet<_> = s1.chars().collect();
        s2.chars()
            .filter(|c| set.contains(c))
            .take(1)
            .for_each(|c: char| total += char_to_val(c));
    });
    total
}

pub fn print_solution(test_val: u32) {
    let input = std::fs::read_to_string("./input").unwrap();
    let test = std::fs::read_to_string("./test").unwrap();

    if solution(test) != test_val {
        println!("Test failed");
    } else {
        println!("{}", solution(input));
    }
}
